app/models/manageiq/providers/openstack/inventory/parser/network_manager.rb

```diff
@@ -48,6 +48,10 @@ def cloud_networks
         subnet.parent_cloud_subnet = persister.cloud_subnets.lazy_find(s.attributes["vsd_managed"])
         subnet.cloud_tenant = persister.cloud_tenants.lazy_find(s.tenant_id)
         subnet.cloud_network = network
+        # Always default this to nil so that if a interface that connects this
+        # to a router are removed, then the association is properly broken when
+        # the subnet is refreshed.
+        subnet.network_router = nil
```

Comment:
Just quick question @mansam, where is the code which connects subnet to a network router?

Reply:
hm, this might not be the best. I think that targeted refresh would be then nullifying the relation (unless it always refreshes together with the interface)

Best would be to just define dependent => :nullify on the relation? That way if router is deleted, the relation disappears. But we probably can't do that on the network_port? Hm we could do custom disconnect method that would nullify it when port is being deleted.

Reply:
It doesn't seem like we could do dependent => :nullify since the port is the thing that gets deleted, but the relationship that has to change is the subnet -> router relationship. I looked into using a custom destroy hook, but `save_inventory_multi` uses `delete` so the destroy hook never gets called.

Reply:
right, save_inventory_multi is for old refresh (that doesn't have targeted refresh)

for having custom delete method you need to specify it like https://github.com/Ladas/manageiq/blob/0815e847bd63282129eee90869bf9bf321c0224d/app/models/manager_refresh/inventory_collection_default.rb#L7